breaking/entrypoint.sh

```diff
@@ -68,11 +71,13 @@ echo "flags: $flags"
 delimiter=$(cat /proc/sys/kernel/random/uuid | tr -d '-')
 echo "breaking<<$delimiter" >>"$GITHUB_OUTPUT"
 
+set +e # don't exit on error running oasdiff can return non-zero exit code, but we just want to capture the output
 if [ -n "$flags" ]; then
-    output=$(oasdiff breaking "$base" "$revision" $flags)
+    output="$(oasdiff breaking "$base" "$revision" $flags)"
 else
-    output=$(oasdiff breaking "$base" "$revision")
+    output="$(oasdiff breaking "$base" "$revision")"
 fi
+set -e # exit on error again
```

Comment:
Could you please describe the use-case for when oasdiff exits with a non-zero code? Additionally, did this occur in issue https://github.com/oasdiff/oasdiff-action/issues/46?

Reply:
oasdiff exits with a non-zero exit code if it detects breaking changes. see https://github.com/Tufin/oasdiff?tab=readme-ov-file#fail-with-exit-code-1-if-any-err-level-breaking-changes-are-found

i believe this was not an issue before because oasdiff was piped into head which overwrote its exit code.

> Additionally, did this occur in issue https://github.com/oasdiff/oasdiff-action/issues/46?

this was the original issue which caused it to print the delimiter error. it would exit after this block and never write the second delimiter

Reply:
You can set `fail-on-diff: false` as mentioned in [this comment](https://github.com/oasdiff/oasdiff-action/issues/46#issuecomment-1988336293).

From what I gather, the issue pertains to the output log and output summary. It seems that this solution may not resolve it.

Reply:
Again, I want it to fail and I want my error messages. This achieves both and gives me the same functionality as v0.0.16 did as you can see here: https://github.com/SCjona/oasdiff-test/actions/runs/8233953647/job/22519631485?pr=1

Reply:
If you set `fail-on-diff: false` you'll not run into this issue as you don't get the non-zero exit code. Since you're running oasdiff twice in this workflow you may consider never passing the `--fail-on` flag on the first invocation, then you wont need the `set +/-x` stuff.

Reply:
If you want we can continue to discuss offline.

Reply:
Also to be considered: What if files can't be found or the network request for the files (if the file is an URL) fails. Then this would also exit with an non-zero code (or so I'd assume) even if you don't pass `--fail-on` to the invocation.

Reply:
Understood, it is a different issue, and there is a beginning of a fix in https://github.com/oasdiff/oasdiff-action/pull/37.